main/test/ca/uwaterloo/flix/library/TestMutSet.flix

```diff
@@ -13,139 +13,147 @@
-        let s2 = MutSet.empty(): MutSet[Int32];
+    def sameElements01(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r): MutSet[Int32, Pure];
+        let s2 = MutSet.emptyWithRegion(r): MutSet[Int32, Pure];
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements02(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements02(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements03(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements03(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(2, s2);
 
         not MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements04(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements04(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
         MutSet.add!(2, s);
         MutSet.add!(3, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(2, s2);
         MutSet.add!(3, s2);
         MutSet.add!(1, s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements05(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements05(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!("a", s);
         MutSet.add!("b", s);
         MutSet.add!("c", s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!("c", s2);
         MutSet.add!("a", s2);
         MutSet.add!("b", s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def setSameElements06(): Bool & Impure =
-        let s = MutSet.empty();
+    def setSameElements06(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!("a", s);
         MutSet.add!("b", s);
         MutSet.add!("c", s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!("c", s2);
         MutSet.add!("a", s2);
 
         not MutSet.sameElements(s, s2)
+    }
 
 
     /////////////////////////////////////////////////////////////////////////////
     // sum                                                                     //
     /////////////////////////////////////////////////////////////////////////////
 
     @test
-    def sum01(): Bool & Impure =
-        MutSet.empty() |> MutSet.sum == 0
+    def sum01(): Bool = region r {
+        MutSet.emptyWithRegion(r) |> MutSet.sum == 0
+    }
 
     @test
-    def sum02(): Bool & Impure =
+    def sum02(): Bool =
         Set#{1} |> Set.toMutSet |> MutSet.sum == 1
```

Comment:
Maybe instead of converting Set to MutSet we should build the set organically and not depend on the Set detour.